src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+                   write(iulog, *) g, il, c, col%landunit(c)
+                   call endrun('something wrong')
+                end if
+                if (.not. any(c==filter_soilc(1:num_soilc))) then
+                   write(iulog, *) c, n_manure_spread_col(c)
+                   call endrun('column not in soilfilter')
+                end if
+
+                if (crop_man_is4crop_area) then
+                   invscale = 1.0_r8
+                else
+                   invscale = 1.0_r8 / lun%wtgcell(l)
+                end if
+
+                n_manure_mixed_col(c) = (ndep_ngrz_grc(g) + ndep_sgrz_grc(g)) * kg_to_g * invscale
+
+                tempr_min_10day = temperature_inst%t_a10min_patch(col%patchi(c))
+                if (tempr_min_10day > tempr_min_grazing) then
+                   ! fraction of animals grazing -> allocate some manure to grasslands before barns
+                   flux_grazing = max_grazing_fract * ndep_sgrz_grc(g) * kg_to_g * invscale
+                   flux_avail_rum = (ndep_sgrz_grc(g)*(1.0_r8 - max_grazing_fract)) * kg_to_g * invscale
+                   grz_fract(c) = max_grazing_fract
+                else
+                   flux_grazing = 0.0_r8
+                   flux_avail_rum = ndep_sgrz_grc(g) * kg_to_g * invscale
+                   grz_fract(c) = 0.0_r8
+                end if
+                flux_avail_mg = ndep_ngrz_grc(g) * kg_to_g * invscale
+                flux_grass_graze = flux_grass_graze + flux_grazing*col%wtgcell(c)
+
+                if (flux_avail_rum > 1e12 .or. flux_avail_mg > 1e12 .or. isnan(flux_avail_mg) .or. isnan(flux_avail_rum)) then
+                   write(iulog, *) 'bad flux_avail', ndep_ngrz_grc(g), ndep_sgrz_grc(g), lun%wtgcell(l)
+                   call endrun('bad flux_avail')
+                end if
+
+                totalinput = totalinput + flux_avail_rum + flux_avail_mg
+
+                counter = 0
+                if (col_grass == c) call endrun('Something wrong with the indices')
+                if (col%patchi(c) /= col%patchf(c)) then
+                   call endrun(msg="ERROR crop column has multiple patches")
+                end if
+
+                tempr_ave = temperature_inst%t_ref2m_patch(col%patchi(c))
+                windspeed_ave = frictionvel_inst%u10_patch(col%patchi(c))
+
+                man_n_barns(c) = flux_avail_rum + flux_avail_mg
+                
+                ! Evaluate the NH3 losses, separate for ruminants (open barns) and others
+                ! (poultry and pigs, closed barns). Note the slicing of fluxes(:,:) and fluxes_tan(:,:).
+                nh3_flux_stores(c) = 0.0
+
+                if (flux_avail_rum < 0) then 
+                   write(iulog, *) 'flux:', flux_avail_rum
+                   call endrun(msg='negat flux_avail for ruminants')
+                end if
+
+                ! Ruminants
+                call eval_fluxes_storage(flux_avail_rum, 'open', tempr_ave, windspeed_ave, 0.0_r8, &
```

Comment:
Here and in many other places: please use keyword=argument syntax when passing arguments to functions and subroutines that have more than a couple of inputs, to avoid the possibility that the wrong order is used.

Reply:
This hasn't been done, but I won't hold up the PR for it.